src/01-dinosaur-facts.js

```diff
@@ -71,7 +112,30 @@ function getDinosaurDescription(dinosaurs, id) {}
  *  getDinosaursAliveMya(dinosaurs, 65, "unknown-key");
  *  //> ["WHQcpcOj0G"]
  */
-function getDinosaursAliveMya(dinosaurs, mya, key) {}
+
+// This function is returning an array of dinosaurs who were alive at the given 'mya'
+// If the mya key is provided, RETURN the value of that key for each dinosaur alive at that time
+// Otherwise return ID
+
+// Delcare a variable that holds an empty array
+// Create a for loop that loops through each element
+// Create an if statement to grab dinosaurs that have the same mya year
+// If the dinosaur only has a singly mya year, return that specific value
+
+function getDinosaursAliveMya(dinosaurs, mya, key) {
+  //Define your default value
+  let dino = [];
+
+  //Define your loop
+  for (const i of dinosaurs) {
+    if (i.mya[i.mya.length - 1] === mya || i.mya - 1 === mya) {
+      dino.push(i[key]);
+    } else if (i.mya[i.mya.length - 1] === mya || i.mya - 1 === mya) {
```

Comment:
The conditionals for both of these if statements are the same. Was that a mistake?